### PR TITLE
Fix `ShiftedBetaGeoModel` broadcasting bug in predictive methods

### PR DESCRIPTION
**Pull request by ColtAllen**
## Description

Fixing another bug I found while working on the tutorial notebook for the newest CLV model. It also fixes a `mypy` check that came up in `customer_choice/mv_its.py`

<details>
<summary>Toggle <b>Product Requirements Document</b></summary>

# Fix ShiftedBetaGeoModel Cohort Broadcasting Bug (Covariate Path)

## Problem Statement

When using `ShiftedBetaGeoModel` with covariates (`dropout_covariate_cols`), predictions for a single cohort differ depending on whether:

- **Scenario A (All Cohorts)**: `expected_probability_alive()` is called with `data=None` (uses all cohorts), then `.sel(cohort=X)` is applied
- **Scenario B (Single Cohort)**: `expected_probability_alive(data=single_cohort_df)` is called with pre-filtered single-cohort data

**Expected behavior**: Both scenarios should produce identical results for the same cohort.

**Actual behavior**: Mean values and HDI bands differ between scenarios.

**Scope**: Bug only affects models with covariates; non-covariate models work correctly.

## Root Cause Analysis

The bug is in [`_extract_predictive_variables`](pymc_marketing/clv/models/shifted_beta_geo.py) lines 403-444 (covariate path). Three potential issues:

### Issue 1: Missing `customer_id` Coordinates on `.isel()` Result (Lines 428-434)

```python
alpha_pred = alpha_cohort.isel(
    cohort=xarray.DataArray(pred_cohort_idx, dims="customer_id")  # NO coords!
) * np.exp(...)
```

The `xarray.DataArray(pred_cohort_idx, dims="customer_id")` has NO coordinate values, causing the `.isel()` result to lose `customer_id` coordinates. When multiplied with `exp(xarray.dot(...))` which HAS coordinates, xarray alignment may behave inconsistently.

**Compare to non-covariate path (lines 451-458)** which correctly uses `.sel()` with a DataArray that HAS coordinates:

```python
customer_cohort_mapping = xarray.DataArray(
    customer_cohort_map.values,
    dims=("customer_id",),
    coords={"customer_id": customer_cohort_map.index},  # HAS coords
)
alpha_pred = alpha_cohort.sel(cohort=customer_cohort_mapping)  # .sel() not .isel()
```

### Issue 2: `cohorts_present` Order Dependency (Lines 226-227)

```python
cohorts_present = pd.Index(data["cohort"].unique())
cohorts_present = cohorts_present.intersection(pd.Index(self.cohorts))
```

The `intersection()` return order is not guaranteed. When `pd.Categorical(..., categories=cohorts_present).codes` computes positional indices, the order must match the order of `alpha_cohort`'s cohort dimension.

### Issue 3: Inconsistent Coordinate Propagation After Multiplication

The covariate path multiplies:

- `alpha_cohort.isel(...)` - dims ("chain", "draw", "customer_id"), NO customer_id coord
- `exp(xarray.dot(dropout_xarray, ...))` - dims ("chain", "draw", "customer_id"), HAS customer_id coord

xarray's broadcasting rules may handle this inconsistently across different array sizes (all cohorts vs single cohort).

## Proposed Fix

Align the covariate path with the non-covariate path by using label-based selection (`.sel()`) instead of position-based (`.isel()`), and ensure all intermediate DataArrays have proper coordinates.

### Changes to [`pymc_marketing/clv/models/shifted_beta_geo.py`](pymc_marketing/clv/models/shifted_beta_geo.py)

Replace lines 422-444 with coordinate-aware selection:

```python
# Create customer-cohort mapping with coordinates (matching non-covariate path)
customer_cohort_mapping = xarray.DataArray(
    customer_cohort_map.values,
    dims=("customer_id",),
    coords={"customer_id": customer_cohort_map.index},
    name="customer_cohort_mapping",
)

# Use .sel() for label-based selection (not .isel())
alpha_base = alpha_cohort.sel(cohort=customer_cohort_mapping)
beta_base = beta_cohort.sel(cohort=customer_cohort_mapping)

# Apply covariate effects
covariate_effect_alpha = np.exp(
    -xarray.dot(dropout_xarray, dropout_coefficient_alpha, dim="dropout_covariate")
)
covariate_effect_beta = np.exp(
    -xarray.dot(dropout_xarray, dropout_coefficient_beta, dim="dropout_covariate")
)

alpha_pred = alpha_base * covariate_effect_alpha
alpha_pred.name = "alpha"

beta_pred = beta_base * covariate_effect_beta
beta_pred.name = "beta"
```

## Unit Tests

Add to [`tests/clv/models/test_shifted_beta_geo.py`](tests/clv/models/test_shifted_beta_geo.py):

### Test 1: Single-Cohort vs All-Cohorts Consistency

```python
def test_covariate_predictions_single_vs_all_cohorts_consistency(self):
    """Predictions for a cohort must be identical whether queried via
    all-cohorts or single-cohort data."""
    train_data = pd.DataFrame({
        "customer_id": range(60),
        "recency": [3, 4, 5, 5, 5, 5] * 10,  # Mix of churned and active
        "T": [5] * 60,
        "cohort": ["A"] * 20 + ["B"] * 20 + ["C"] * 20,
        "channel": [0, 1] * 30,
    })

    model = ShiftedBetaGeoModel(
        data=train_data,
        model_config={"dropout_covariate_cols": ["channel"]}
    )
    model.fit(method="map", maxeval=50)

    # Get active customers for predictions
    active_all = train_data.query("recency == T").copy()
    active_B = active_all[active_all["cohort"] == "B"].copy()

    # Scenario A: Predict all cohorts, select cohort B
    pred_all = model.expected_probability_alive(data=active_all, future_t=2)
    pred_all_B = pred_all.sel(cohort="B")

    # Scenario B: Predict single cohort B directly  
    pred_single_B = model.expected_probability_alive(data=active_B, future_t=2)

    # Values must be identical (within floating point tolerance)
    xr.testing.assert_allclose(
        pred_all_B.mean("cohort"),  # Average over customers in cohort B
        pred_single_B.mean("cohort"),
        rtol=1e-10
    )
```

### Test 2: Covariate Effect Consistency Across Query Methods

```python
def test_covariate_effect_preserved_single_cohort(self):
    """Covariate effects must be identical regardless of query method."""
    train_data = pd.DataFrame({
        "customer_id": range(40),
        "recency": [4, 4, 4, 4] * 10,  # All active
        "T": [4] * 40,
        "cohort": ["X"] * 20 + ["Y"] * 20,
        "feature": [0.0, 1.0] * 20,  # Binary covariate
    })

    model = ShiftedBetaGeoModel(
        data=train_data,
        model_config={"dropout_covariate_cols": ["feature"]}
    )
    model.fit(method="map", maxeval=50)

    active = train_data.query("recency == T").copy()
    active_X = active[active["cohort"] == "X"].copy()

    # All cohorts prediction
    dataset_all = model._extract_predictive_variables(
        active.assign(future_t=1),
        customer_varnames=["T", "future_t", "cohort"]
    )

    # Single cohort prediction
    dataset_single = model._extract_predictive_variables(
        active_X.assign(future_t=1),
        customer_varnames=["T", "future_t", "cohort"]
    )

    # Alpha values for cohort X customers must match
    alpha_all_X = dataset_all["alpha"].sel(cohort="X")
    alpha_single_X = dataset_single["alpha"]

    xr.testing.assert_allclose(alpha_all_X, alpha_single_X, rtol=1e-10)
```

### Test 3: Intermediate Coordinate Integrity

```python
def test_covariate_path_coordinate_integrity(self):
    """Verify customer_id coordinates are preserved through covariate path."""
    train_data = pd.DataFrame({
        "customer_id": [100, 101, 102, 103],
        "recency": [3, 3, 3, 3],
        "T": [3] * 4,
        "cohort": ["P", "P", "Q", "Q"],
        "cov": [0.5, 1.5, 0.5, 1.5],
    })

    model = ShiftedBetaGeoModel(
        data=train_data,
        model_config={"dropout_covariate_cols": ["cov"]}
    )
    model.fit(method="map", maxeval=10)

    pred_data = train_data.query("recency == T").assign(future_t=1)
    dataset = model._extract_predictive_variables(
        pred_data, customer_varnames=["T", "future_t", "cohort"]
    )

    # Verify customer_id coordinate exists and matches input
    # (After swap_dims, customer_id becomes a coord on cohort dim)
    assert "customer_id" in dataset.coords or "cohort" in dataset.dims
    
    # Verify alpha has proper dimensions
    assert "alpha" in dataset
    assert dataset["alpha"].dims[-1] == "cohort"  # After swap_dims
```

## Validation Criteria

1. All existing tests pass
2. New tests pass
3. The user's notebook produces identical plots for both scenarios
4. Predictions for any cohort are identical whether queried via all-cohorts or single-cohort data
</details

## Related Issue

- [x] Closes https://github.com/pymc-labs/pymc-marketing/issues/2114

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://www.pymc-marketing.io/en/latest/contributing/index.html). Feel free to comment [`pre-commit.ci autofix` to auto-fix](https://pre-commit.ci/#configuration-autofix_prs).
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks) using [numpydoc format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

----
📚 Documentation preview 📚: https://pymc-marketing--2121.org.readthedocs.build/en/2121/

### CODE REVIEW

**Review by juanitorduz — APPROVED** [review]
Thanks @ColtAllen
